src/sys/signal.rs

```diff
@@ -488,102 +489,114 @@ pub enum SigevNotify {
                     si_value: libc::intptr_t },
 }
 
-/// Used to request asynchronous notification of the completion of certain
-/// events, such as POSIX AIO and timers.
-#[repr(C)]
-pub struct SigEvent {
-    sigevent: libc::sigevent
-}
+#[cfg(not(target_os = "openbsd"))]
+pub mod sigevent {
```

Comment:
I don't think this module should be public since you glob-import everything into the `sys::signal` module above.